mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/ar/ArChessBoardScreen.kt

```diff
@@ -68,22 +65,60 @@ fun <Piece : ChessBoardState.Piece> ArChessBoardScreen(
  * and change the displayed position
  *
  * @param arSceneView The view where the scene will be displayed
- * @param chessScene The scene that contains the boards node
+ * @param boardNode The node that will be added into the view
  * @param anchor The (new) board's anchor position
  */
-private fun <Piece : ChessBoardState.Piece> anchorOrMoveBoard(
-    arSceneView: ArSceneView,
-    chessScene: ChessScene<Piece>?,
+private fun anchorOrMoveBoard(
+    // arSceneView: ArSceneView,
+    boardNode: ArModelNode,
     anchor: Anchor
 ) {
 
-  val currentChessScene = chessScene ?: return
+  // FIXME : Workaround see line 55
+
+  // Add only one instance of the node
+  /*if (!arSceneView.children.contains(boardNode)) {
+    arSceneView.addChild(boardNode)
+  }*/
+
+  if (!boardNode.isVisible) boardNode.isVisible = true
+
+  boardNode.anchor = anchor
+}
+
+/**
+ * Create an instance of [ChessScene] and setup the onTouch callback
+ *
```

Comment:
Missing documentation for `Piece` type parameter.

Reply:
Fix in https://github.com/epfl-SDP/android/pull/415/commits/3c79295548979f5ffd972831dafda98d86b50618